airbyte-integrations/connectors/source-bing-ads/source_bing_ads/source.py

```diff
@@ -83,17 +86,66 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
         try:
             client = Client(**config)
             account_ids = {str(account["Id"]) for account in Accounts(client, config).read_records(SyncMode.full_refresh)}
+            self.validate_custom_reposts(config, client)
             if account_ids:
                 return True, None
             else:
                 raise AirbyteTracedException(
-                    message="Config validation error: You don't have accounts assigned to this user.",
+                    message="Config validation error: You don't have accounts assigned to this user. Please verify your developer token.",
                     internal_message="You don't have accounts assigned to this user.",
                     failure_type=FailureType.config_error,
                 )
         except Exception as error:
             return False, error
 
+    def validate_custom_reposts(self, config: Mapping[str, Any], client: Client):
+        custom_reports = self.get_custom_reports(config, client)
+        for custom_report in custom_reports:
+            try:
+                for account in Accounts(client, config).read_records(SyncMode.full_refresh):
+                    list(
+                        custom_report.read_records(
+                            sync_mode=SyncMode.full_refresh,
+                            stream_slice={"account_id": account["Id"], "customer_id": account["ParentCustomerId"]},
+                        )
+                    )
```

Comment:
Check connection will do full sync here?
We can use parsed `/bingads/v13/proxies/production/reporting_service.xml` to validate input data (and/or try to use `enum` values from there).
The only potential errors will left and should be handled during the read:
- combination of columns (see Column restrictions) 
- combination of columns with report_aggregation

Reply:
updated, now it  validates object name and columns, and combination of columns with report_aggregation validates while read